boards/FSM/Test/Src/Test_StateMachine.cpp

```diff
@@ -481,48 +559,223 @@ TEST_F(FsmStateMachineTest, exit_air_closed_state_when_air_negative_is_opened)
+
+        is_brake_actuated_fake.return_val = true;
+
+        LetTimePass(state_machine, 9);
+        ASSERT_EQ(
+            50, App_CanTx_GetPeriodicSignal_MAPPED_PEDAL_PERCENTAGE(
+                    can_tx_interface));
+        LetTimePass(state_machine, 1);
+        ASSERT_EQ(
+            0, App_CanTx_GetPeriodicSignal_MAPPED_PEDAL_PERCENTAGE(
+                   can_tx_interface));
+    }
+}
+
+// FSM-6
+TEST_F(
+    FsmStateMachineTest,
+    papps_greater_than_apps_by_ten_percent_sets_motor_shutdown_flag)
+{
+    CheckIfAppsDisagreementFaultIsSetWhenAppsHasDisagreement(
+        get_papps_encoder_counter_fake.return_val,
+        get_sapps_encoder_counter_fake.return_val,
+        App_CanTx_GetPeriodicSignal_APPS_HAS_DISAGREEMENT,
+        PAPPS_MAX_PRESSED_VALUE, SAPPS_MAX_PRESSED_VALUE,
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_APPS_HAS_DISAGREEMENT_TRUE_CHOICE,
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_APPS_HAS_DISAGREEMENT_FALSE_CHOICE);
+}
+
+// FSM-6
+TEST_F(
+    FsmStateMachineTest,
+    sapps_greater_than_apps_by_ten_percent_sets_motor_shutdown_flag)
+{
+    CheckIfAppsDisagreementFaultIsSetWhenAppsHasDisagreement(
+        get_sapps_encoder_counter_fake.return_val,
+        get_papps_encoder_counter_fake.return_val,
+        App_CanTx_GetPeriodicSignal_APPS_HAS_DISAGREEMENT,
+        SAPPS_MAX_PRESSED_VALUE, PAPPS_MAX_PRESSED_VALUE,
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_APPS_HAS_DISAGREEMENT_TRUE_CHOICE,
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_APPS_HAS_DISAGREEMENT_FALSE_CHOICE);
+}
 
-    // Start with a non-zero pedal position to prevent false positive
-    App_CanTx_SetPeriodicSignal_SAPPS_MAPPED_PEDAL_PERCENTAGE(
-        can_tx_interface, 50.0f);
+// FSM-7
+TEST_F(
+    FsmStateMachineTest,
+    apps_is_pressed_and_brake_is_not_actuated_does_not_set_motor_shutdown_flag)
+{
+    const float fake_encoder_value_threshold =
+        GetEncoderCounterFromMappedPedalPercentage(
+            25.0, PAPPS_MAX_PRESSED_VALUE);
+    is_brake_actuated_fake.return_val = false;
+
+    // Check CAN signals when the mapped pedal percentage is less than 25%
+    get_papps_encoder_counter_fake.return_val =
+        fake_encoder_value_threshold - 1;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_PLAUSIBILITY_CHECK_HAS_FAILED_FALSE_CHOICE,
+        App_CanTx_GetPeriodicSignal_PLAUSIBILITY_CHECK_HAS_FAILED(
+            can_tx_interface));
+
+    // Check CAN signals when the mapped pedal percentage is greater than 25%
+    get_papps_encoder_counter_fake.return_val =
+        fake_encoder_value_threshold + 1;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(
+        CANMSGS_FSM_MOTOR_SHUTDOWN_ERRORS_PLAUSIBILITY_CHECK_HAS_FAILED_FALSE_CHOICE,
+        App_CanTx_GetPeriodicSignal_PLAUSIBILITY_CHECK_HAS_FAILED(
+            can_tx_interface));
+}
+
+// FSM-7
+TEST_F(
+    FsmStateMachineTest,
+    apps_is_pressed_and_brake_is_actuated_sets_motor_shutdown_flag)
```

Comment:
might want to change `flag` to `can_tx_signal` for this test title and others

Reply:
Okay yeah this is more descriptive. Fixed